### PR TITLE
Solpio/UI/ra 27

### PR DESCRIPTION
**Pull request by Solpio**

### CODE REVIEW

File: admin-app/src/services/api/index.ts [hunk 1]
```diff
@@ -3,15 +3,26 @@ import { config } from "./config";
...
-		const chatTableData = await instance.get<T>(path);
+		const chatTableData = await instance.get<T>(path, {
+			headers: {
+				Authorization: `Bearer ${token}`,
+			},
+		});
 		return chatTableData.data;
 	} catch (error) {
 		if (axios.isAxiosError(error)) {
+			if (error.response) {
```
Comment (kurkul608) [inline admin-app/src/services/api/index.ts]: remove

File: admin-app/src/services/api/index.ts [hunk 2]
```diff
@@ -3,15 +3,26 @@ import { config } from "./config";
...
+		"Content-Type": "application/json",
+	},
 });
 
-export const get = async <T>(path: string) => {
+export const get = async <T>(path: string, token: string) => {
 	try {
-		const chatTableData = await instance.get<T>(path);
+		const chatTableData = await instance.get<T>(path, {
+			headers: {
```
Comment (kurkul608) [inline admin-app/src/services/api/index.ts:17]: create func getHeaders
path, {
			...getHeaders(authToken)
}

File: admin-app/src/services/api/index.ts [hunk 3]
```diff
@@ -3,15 +3,26 @@ import { config } from "./config";
...
 export const instance = axios.create({
 	...config.api,
-	headers: { "Content-Type": "application/json" },
+	headers: {
+		"Content-Type": "application/json",
+	},
 });
 
-export const get = async <T>(path: string) => {
+export const get = async <T>(path: string, token: string) => {
```
Comment (kurkul608) [inline admin-app/src/services/api/index.ts]: change name to authToken

File: admin-app/src/services/api/index.ts [hunk 4]
```diff
@@ -25,6 +36,11 @@ export const post = async <T, D>(path: string, body: D) => {
 		return chatTableData.data;
 	} catch (error) {
 		if (axios.isAxiosError(error)) {
+			if (error.response) {
```
Comment (kurkul608) [inline admin-app/src/services/api/index.ts]: remove

File: admin-app/src/shared/auth/components/auth-form/index.tsx [hunk 5]
```diff
@@ -9,15 +9,16 @@ import {
 } from "./styled";
 // import { Title } from "./styled";
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-form/index.tsx]: please, remove ALL comments in this file

File: admin-app/src/shared/auth/components/auth-form/index.tsx [hunk 6]
```diff
@@ -9,15 +9,16 @@ import {
...
 import { useNavigate } from "react-router";
+import { getAuthToken } from "helpers/storage-parser";
 // import { useFormik } from "formik";
 // import * as yup from "yup";
 
 export const AuthForm = () => {
-	const { token } = useAppSelector((state) => state.auth);
-	const isAuth = !!token;
+	const token = getAuthToken();
+	const isAuth = !!token.length;
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-form/index.tsx]: why lenght?
token must be "String"
!!("" || undefined || null) -> false
correct version const isAuth = !!getAuthToken()

File: admin-app/src/shared/auth/components/auth-form/index.tsx [hunk 7]
```diff
@@ -9,15 +9,16 @@ import {
...
 // import { authAsync, IUserLogin } from "../../redux/auth.slice";
 import { useNavigate } from "react-router";
+import { getAuthToken } from "helpers/storage-parser";
 // import { useFormik } from "formik";
 // import * as yup from "yup";
 
 export const AuthForm = () => {
-	const { token } = useAppSelector((state) => state.auth);
-	const isAuth = !!token;
+	const token = getAuthToken();
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-form/index.tsx]: if you don't use token, then remove thist const

File: admin-app/src/shared/auth/components/auth-token-widget/index.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import { useAppDispatch } from "redux/hooks";
+import { logIn } from "shared/auth/redux/auth.slice";
+
+export const AuthTokenWidget = () => {
+	const authToken = useParams();
+	const navigate = useNavigate();
+	const dispatch = useAppDispatch();
+	if (authToken.token) {
+		localStorage.setItem("auth", authToken.token);
+		dispatch(logIn(localStorage.getItem("auth")));
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-token-widget/index.tsx]: it seems to me that the result of localStorage.get() will be JSON 
It makes sense to make a function that makes a .get request and decodes the resulting JSON

File: admin-app/src/shared/auth/components/auth-token-widget/index.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+	}
+	setTimeout(() => {
+		return navigate(routeBuilder([Routes.admin]));
+	}, 3000);
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-token-widget/index.tsx]: why setTimeOut?

File: admin-app/src/shared/auth/components/auth-token-widget/index.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+	return (
+		<Widget name={""}>
+			Мужык, ты через 3 секунды авторизуешься )))))00))0)000
```
Comment (kurkul608) [inline admin-app/src/shared/auth/components/auth-token-widget/index.tsx]: remove this

File: admin-app/src/shared/layout/header/logout/index.tsx [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { StyledButton } from "shared/layout/header/logout/styled";
+import { useAppDispatch } from "redux/hooks";
+import { logOut } from "shared/auth/redux/auth.slice";
+
+export const Logout = () => {
+	const dispatch = useAppDispatch();
+	const handleOnClick = () => {
+		dispatch(logOut());
+		localStorage.removeItem("auth");
```
Comment (kurkul608) [inline admin-app/src/shared/layout/header/logout/index.tsx]: please create localStorageService with remove, set, get services

File: admin-app/src/shared/router/index.tsx [hunk 12]
```diff
@@ -1,22 +1,30 @@
...
 import { ChatListPage } from "pages/chat-list-page";
 import { ChatSettings } from "pages/chat-settings";
+import { AuthTokenPage } from "pages/auth-token-page";
+import { useAppSelector } from "redux/hooks";
 
 const NotFound = () => {
 	return <div>Page not found</div>;
 };
 
+const PrivateRoute = () => {
```
Comment (kurkul608) [inline admin-app/src/shared/router/index.tsx]: move PrivateRoute to private-route.tsx

File: admin-app/src/shared/auth/components/auth-token-widget/index.tsx (hunk 8 above)
Comment (Solpio) [reply admin-app/src/shared/auth/components/auth-token-widget/index.tsx]: Return value will be a string or null.
https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem

**Pull request merged**